Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -68,9 +71,26 @@ async function run() {
         webDeployPkg = availableWebPackages[0];
 
         var isFolderBasedDeployment = utility.isInputPkgIsFolder(webDeployPkg);
+        var generateTempFolder = JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution || addWebConfig;
 
+        if(generateTempFolder) {
+            var folderPath = await fileTransformationsUtility.generateTemporaryFolder(isFolderBasedDeployment, webDeployPkg);
```

Comment:
do not move generateTemporaryFolder from utility.ts.
Use utility.generateTempFolder()